src/Elements/ElementBlogOverview.php

```diff
@@ -0,0 +1,511 @@
+            return $this->widgetArea;
+        }
+
+        $page = $this->getPage();
+
+        // There isn't anything we can do if there is no Page. Unlike with BlogPosts, it really doesn't make sense to
+        // return all Widgets in the DB...
+        if ($page === null) {
+            return null;
+        }
+
+        // Check that the parent Page is a Blog page. We also can't get widgets for the Page if the Page type doesn't
+        // have the Widget extension
+        if (!$page instanceof Blog || !$page->hasExtension(WidgetPageExtension::class)) {
+            // You have specified that this Block *cannot* be used outside of the Blog, so that also means that we
+            // expect a parent Page to have been present and of type Blog
+            if (!static::config()->get('allow_use_outside_of_blog')) {
+                return null;
+            }
+
+            // You get one last chance to return a WidgetArea through some other means
+            if ($page->hasMethod('SideBarView')) {
+                // setWidgetArea() requires a class of WidgetArea|null be provided, so, it'll throw exceptions for us if
+                // any other type is provided
+                $this->setWidgetArea($page->SideBarView());
+
+                return $this->widgetArea;
+            }
+
+            return null;
+        }
+
+        // If the Page is inheriting it's SideBar, then we should grab it from the Parent
+        if ($page->InheritSideBar
+            && ($parent = $page->getParent())
+            && $parent->hasMethod('SideBarView')
+        ) {
+            // Store this WidgetArea as our cached value and provide an extension point
+            $this->setWidgetArea($parent->SideBarView());
+
+            return $this->widgetArea;
+        }
+
+        // Otherwise, let's attempt to fetch the SideBar from this current Page
+        if ($page->SideBar()->exists()) {
+            // Store this WidgetArea as our cached value and provide an extension point
+            $this->setWidgetArea($page->SideBar());
+
+            return $this->widgetArea;
+        }
+
+        return null;
+    }
+
+    /**
+     * If you're using partial cache, then use can use this as the cache key for the Block
+     *
+     * @return string|null
+     * @throws ValidationException
+     */
+    public function getCacheKey(): ?string
+    {
+        if ($this->cacheKey !== null) {
+            return $this->cacheKey;
+        }
+
+        $pieces = [
+            static::class,
+            $this->ID,
+            $this->LastEdited,
+        ];
+
+        if ($this->getBlogPosts() !== null) {
+            $pieces[] = $this->getBlogPosts()->count();
+            $pieces[] = $this->getBlogPosts()->max('LastEdited');
+        }
+
+        $cacheKey = implode('-', $pieces);
+
+        $this->invokeWithExtensions('updateCacheKey', $cacheKey);
+
+        $this->cacheKey = $cacheKey;
+
+        return $this->cacheKey;
+    }
```

Comment:
Making a note for myself for later:

Remove this. It seems useful, but it conflicts with other modules that provide a `getCacheKey()` method through extension. I think devs are pretty familiar with creating their own cache keys, so we prob don't need to provide this.